TSPL.docc/LanguageGuide/Concurrency.md

```diff
@@ -255,91 +256,107 @@ only certain places in your program can call asynchronous functions or methods:
   - Code at the top level that forms an implicit main function.
 -->
 
-Code in between possible suspension points runs sequentially,
-without the possibility of interruption from other concurrent code.
-For example, the code below moves a picture from one gallery to another.
+You can explicitly insert a suspension point
+by calling the [`Task.yield()`][] method.
 
-```swift
-let firstPhoto = await listPhotos(inGallery: "Summer Vacation")[0]
-add(firstPhoto, toGallery: "Road Trip")
-// At this point, firstPhoto is temporarily in both galleries.
-remove(firstPhoto, fromGallery: "Summer Vacation")
-```
-
-There's no way for other code to run in between
-the call to `add(_:toGallery:)` and `remove(_:fromGallery:)`.
-During that time, the first photo appears in both galleries,
-temporarily breaking one of the app's invariants.
-To make it even clearer that this chunk of code
-must not have `await` added to it in the future,
-you can refactor that code into a synchronous function:
+[`Task.yield()`]: https://developer.apple.com/documentation/swift/task/3814840-yield
 
 ```swift
-func move(_ photoName: String, from source: String, to destination: String) {
-    add(photoName, toGallery: destination)
-    remove(photoName, fromGallery: source)
+func generateSlideshow(forGallery gallery: String) async {
+    let photos = await listPhotos(inGallery: gallery)
+    for photo in photos {
+        // ... render a few seconds of video for this photo ...
+        await Task.yield()
+    }
 }
-// ...
-let firstPhoto = await listPhotos(inGallery: "Summer Vacation")[0]
-move(firstPhoto, from: "Summer Vacation", to: "Road Trip")
 ```
 
-In the example above,
-because the `move(_:from:to:)` function is synchronous,
-you guarantee that it can never contain possible suspension points.
-In the future,
-if you try to add concurrent code to this function,
-introducing a possible suspension point,
-you'll get compile-time error instead of introducing a bug.
+Assuming the code that renders video is synchronous,
+it doesn't contain any suspension points.
+However, that work might take a long time.
+Periodically calling `Task.yield()` explicitly adds suspension points,
+which let Swift balance between making progress on this work,
+and letting other tasks make progress on other work.
```

Comment:
```suggestion
while letting other tasks make progress on other work.
```

Would be good if you could add an example of what that "other work" might be to the narrative.